go/service/usersearch.go

```diff
@@ -188,33 +197,159 @@ func contactSearch(mctx libkb.MetaContext, arg keybase1.UserSearchArg) (res []ke
 	return res, nil
 }
 
+var nonDigits = regexp.MustCompile("[^\\d]")
+
+func imptofuQueryToAssertion(typ keybase1.ImpTofuSearchType, val string) string {
+	switch typ {
+	case keybase1.ImpTofuSearchType_PHONE:
+		return fmt.Sprintf("%s@phone", nonDigits.ReplaceAllString(val, ""))
+	case keybase1.ImpTofuSearchType_EMAIL:
+		return fmt.Sprintf("[%s]@email", strings.ToLower(val))
+	default:
+		panic("invalid keybase1.ImpTofuSearchType enum value")
+	}
+}
+
+func imptofuSearch(mctx libkb.MetaContext, provider contacts.ContactsProvider, imptofuQuery keybase1.ImpTofuQuery) (res *keybase1.APIUserSearchResult, err error) {
+	var emails []keybase1.EmailAddress
+	var phones []keybase1.RawPhoneNumber
+
+	imptofuType, err := imptofuQuery.T()
+	if err != nil {
+		return res, err
+	}
+
+	var queryString string
+
+	switch imptofuType {
+	case keybase1.ImpTofuSearchType_EMAIL:
+		email := imptofuQuery.Email()
+		queryString = string(email)
+		emails = append(emails, email)
+	case keybase1.ImpTofuSearchType_PHONE:
+		phone := keybase1.RawPhoneNumber(imptofuQuery.Phone())
+		queryString = string(phone)
+		phones = append(phones, phone)
+	}
+
+	lookupRes, err := provider.LookupAll(mctx, emails, phones, keybase1.RegionCode(""))
```

Comment:
i guess no luck on the regioncode? i thought ios had this info available for localization?

Reply:
although I guess it's not strictly related

Reply:
it's handled on the frontend entirely - frontend knows the carrier region code and can transform inputs that are ultimately sent to this RPC. one day we will do a cleanup pass to remove this stuff from service.